xrpl/models/requests/custom_request.py

```diff
@@ -0,0 +1,86 @@
+"""A generic request that can be used for unsupported requests."""
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import Any, Dict, Type, Union, cast
+
+from xrpl.models.exceptions import XRPLModelException
+from xrpl.models.requests.request import Request, RequestMethod
+from xrpl.models.utils import require_kwargs_on_init
+
+
+@require_kwargs_on_init
+@dataclass(init=False, frozen=True)
+class CustomRequest(Request):
+    """
+    A request object representing all unsupported requests.
+
+    There is no analog in rippled - this is an xrpl-py-specific model.
+    """
+
+    method: RequestMethod = field(default=RequestMethod.CUSTOM_REQUEST, init=False)
+    """
+    This field is required.
+
+    :meta hide-value:
```

Comment:
doesn't this have to be `init=True`?

Reply:
No - the init of this method doesn't need it. It's only called in the super init, and `Request` does take a method param in its init.